src/Service/EmailService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace App\Service;
+
+use App\Entity\User;
+use Aws\Ses\SesClient;
+use Aws\Exception\AwsException;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class EmailService {
+
+    /**
+     * @var \Swift_Mailer
+     *
+     */
+    private $mailer;
+    private $templating;
+    private $manager;
+
+    public function __construct(\Swift_Mailer $mailer, \Twig_Environment $templating,ObjectManager $manager)
+    {
+        $this->mailer = $mailer;
+        $this->templating = $templating;
+        $this->manager = $manager;
+    }
+
+    //Email function for sending the confirmation email
+    public function confirmRegistration($username, $email, $token)
+    {
+        $message = (new \Swift_Message('Confirmez votre adresse email'))
+            ->setFrom(getenv('MAILER_FROM'))
+            ->setTo($email)
+            ->setBody(
+                $this->templating->render(
+                    'emails/registration.html.twig',
+                    ['username' => $username,
+                        'token' => $token]
+                ),
+                'text/html'
+            );
+        $this->mailer->send($message);
+    }
+
+    public function sendNewEmail($email) {
+
+        $user = $this->manager->getRepository(User::class)
+            ->findOneBy(array('email' => $email));
+
+        $token = $this->gen_uuid();
+        $user->setToken($token);
+        $this->manager->flush();
+
+        $this->confirmRegistration($user->getLogin(), $user->getEmail(), $token);
+    }
+
+    //[NOTE] Keep for later
+    //public function resetPassword($email, $url,  \Swift_Mailer $mailer)
+    //{
+    //    $message = (new \Swift_Message('Mot de passe oublié ?'))
+    //        ->setFrom(getenv('MAILER_FROM'))
+    //        ->setTo($email)
+    //        ->setBody(
+    //            $this->templating->render(
+    //                'emails/forgotPassword.html.twig',
+    //                ['url' => $url]
+    //            ),
+    //            'text/html'
+    //        );
+    //    $this->mailer->send($message);
+    //}
+
+    //Generate a token for the email_token (use in UserController
+    function gen_uuid() {
```

Comment:
Maybe be honest and link the place where you fond this code ;)

Reply:
Done, again, thank you Ivan when you give me this SO thread :p